src/utils/helpers/GraphHelpers.ts

```diff
@@ -188,9 +286,21 @@ export function getHighLightedLabelColor(node: DependencyNode) {
 
 export function handleDrag(simulation: Simulation<DependencyNode, DependencyLink>) {
     return drag<SVGGElement, DependencyNode>()
-        .on('start', (node: DependencyNode) => dragStarted(node, simulation))
-        .on('drag', dragged)
-        .on('end', (node: DependencyNode) => dragEnded(node, simulation));
+        .on('start', (node: DependencyNode) => {
+            if (!selectHighLightedNodes().data().length) {
+                dragStarted(node, simulation);
+            }
+        })
+        .on('drag', (node: DependencyNode) => {
+            if (!selectHighLightedNodes().data().length) {
```

Comment:
this collection probably does not change during drag, so this length could be stored on start, or checked on start only